core/metacling/src/TCling.cxx

```diff
@@ -1138,6 +1141,17 @@ static void LoadCoreModules(cling::Interpreter &interp)
 
    if (!LoadModule(moduleMap.findModule("RIO")->Name, interp))
       Error("TCling::LoadCodeModule", "Cannot load module RIO");
+
+   // Check that the gROOT macro was exported by any core module.
+   assert(interp.getMacro("gROOT") && "Couldn't load gROOT macro?");
+
+   // C99 decided that it's a very good idea to name a macro `I` (the letter I).
+   // This seems to screw up nearly all the template code out there as `I` is
+   // common template parameter name and iterator variable name.
+   // Let's follow the GCC recommendation and undefine `I` in case any of the
+   // core modules have defined it:
+   // https://www.gnu.org/software/libc/manual/html_node/Complex-Numbers.html
+   interp.declare("#ifdef I\n #undef I\n #endif\n");
```

Comment:
@vgvassilev I reply here to [your comment on the closed PR](https://github.com/root-project/root/pull/1378#pullrequestreview-79456335). I don't think we call recursively `LoadCoreModule` or recursively make TCling instances? And if we do, then we already made a good portion of `declare`s before this code for all the other TCling things, so we would anyway be in deep trouble.